src/services/infrastructure/url-generator/url.generator.service.ts

```diff
@@ -288,14 +296,39 @@ export class UrlGeneratorService {
     return '';
   }
 
-  public createUrlForUserNameID(userNameID: string): string {
-    return `${this.endpoint_cluster}/${this.PATH_USER}/${userNameID}`;
+  public createUrlForContributor(contributor: IContributor): string {
+    const type = this.getContributorType(contributor);
+    let path = this.PATH_VIRTUAL_CONTRIBUTOR;
+    switch (type) {
+      case CommunityContributorType.USER:
+        path = this.PATH_USER;
+        break;
+      case CommunityContributorType.ORGANIZATION:
+        path = this.PATH_ORGANIZATION;
+        break;
+      case CommunityContributorType.VIRTUAL:
+        path = this.PATH_VIRTUAL_CONTRIBUTOR;
```

Comment:
already assigned

Reply:
yes, but better to also have here for completeness - not just rely on the original assignment. The other option would be make the original assignment an empty string.